plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,565 @@
+    // Mappings for JDBC column types to internal Trino types
+    private static final Map<Integer, ColumnMapping> STANDARD_COLUMN_MAPPINGS = ImmutableMap.<Integer, ColumnMapping>builder()
+            .put(Types.BOOLEAN, StandardColumnMappings.booleanColumnMapping())
+            .put(Types.TINYINT, StandardColumnMappings.tinyintColumnMapping())
+            .put(Types.SMALLINT, StandardColumnMappings.smallintColumnMapping())
+            .put(Types.INTEGER, StandardColumnMappings.integerColumnMapping())
+            .put(Types.BIGINT, StandardColumnMappings.bigintColumnMapping())
+            .put(Types.REAL, StandardColumnMappings.realColumnMapping())
+            .put(Types.DOUBLE, StandardColumnMappings.doubleColumnMapping())
+            .put(Types.FLOAT, StandardColumnMappings.doubleColumnMapping())
+            .put(Types.BINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .put(Types.VARBINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .put(Types.LONGVARBINARY, StandardColumnMappings.varbinaryColumnMapping())
+            .buildOrThrow();
+
+    private static final Map<String, ColumnMappingFunction> SHOWFLAKE_COLUMN_MAPPINGS = ImmutableMap.<String, ColumnMappingFunction>builder()
+            .put("time", typeHandle -> Optional.of(timeColumnMapping(typeHandle)))
+            .put("timestampntz", typeHandle -> Optional.of(timestampColumnMapping(typeHandle)))
+            .put("timestamptz", typeHandle -> Optional.of(timestampTzColumnMapping(typeHandle)))
+            .put("timestampltz", typeHandle -> Optional.of(timestampTzColumnMapping(typeHandle)))
+            .put("date", typeHandle -> Optional.of(ColumnMapping.longMapping(
+                    DateType.DATE,
+                    (resultSet, columnIndex) -> LocalDate.ofEpochDay(resultSet.getLong(columnIndex)).toEpochDay(),
+                    snowFlakeDateWriter())))
+            .put("object", typeHandle -> Optional.of(ColumnMapping.sliceMapping(
+                    createUnboundedVarcharType(),
+                    StandardColumnMappings.varcharReadFunction(createUnboundedVarcharType()),
+                    StandardColumnMappings.varcharWriteFunction(),
+                    PredicatePushdownController.DISABLE_PUSHDOWN)))
+            .put("array", typeHandle -> Optional.of(ColumnMapping.sliceMapping(
+                    createUnboundedVarcharType(),
+                    StandardColumnMappings.varcharReadFunction(createUnboundedVarcharType()),
+                    StandardColumnMappings.varcharWriteFunction(),
+                    PredicatePushdownController.DISABLE_PUSHDOWN)))
+            .put("variant", typeHandle -> Optional.of(ColumnMapping.sliceMapping(
+                    createUnboundedVarcharType(),
+                    variantReadFunction(),
+                    StandardColumnMappings.varcharWriteFunction(),
+                    PredicatePushdownController.FULL_PUSHDOWN)))
```

Comment:
Remove untested type mappings or add `time`, `object`, `array` & `variant` test cases to TestSnowflakeTypeMapping.